test/syscalls/linux/mq.cc

```diff
@@ -0,0 +1,283 @@
+    }
+  }
+
+  mqd_t fd() { return fd_; }
+
+  const char* name() { return name_.c_str(); }
+
+  mqd_t release() {
+    mqd_t old = fd_;
+    fd_ = -1;
+    return old;
+  }
+
+ private:
+  mqd_t fd_;
+  std::string name_;
+};
+
+// genName generates an unused name to be used by mq_open(2) to create message
+// queues. genName generates the name by choosing a psuedo-random name, and
+// testing if it exists. While it's most likely that a name will be found in
+// the first few iterations, it is possible that genName may loop infinitely.
+std::string genName() {
+  int rand = int(getpid());
+  for (;;) {
+    rand = rand * 1664525 + 1013904223;  // From numerical recipes.
+
+    std::string name = "/" + std::to_string(rand);
+    mqd_t d = mq_open(name.c_str(), O_RDWR);
+    if (d == -1 && errno == ENOENT) {
+      return name;
+    }
+    mq_close(d);
+  }
+}
+
+// MqOpen wraps mq_open(3) using a given name.
+PosixErrorOr<PosixQueue> MqOpen(std::string name, int oflag) {
+  mqd_t fd = mq_open(name.c_str(), oflag);
+  if (fd == -1) {
+    return PosixError(errno, absl::StrFormat("mq_open(%s, %d)", name, oflag));
+  }
+  return PosixQueue(fd, name);
+}
+
+// MqOpen wraps mq_open(3) using a generated name.
+PosixErrorOr<PosixQueue> MqOpen(int oflag) {
+  auto name = genName();
+  mqd_t fd = mq_open(name.c_str(), oflag);
+  if (fd == -1) {
+    return PosixError(errno, absl::StrFormat("mq_open(%d)", oflag));
+  }
+  return PosixQueue(fd, name);
+}
+
+// MqOpen wraps mq_open(3) using a given name.
+PosixErrorOr<PosixQueue> MqOpen(int oflag, mode_t mode, struct mq_attr* attr) {
+  auto name = genName();
+  mqd_t fd = mq_open(name.c_str(), oflag, mode, attr);
+  if (fd == -1) {
+    return PosixError(errno, absl::StrFormat("mq_open(%d)", oflag));
+  }
+  return PosixQueue(fd, name);
+}
+// MqOpen wraps mq_open(3) using a generated name.
+PosixErrorOr<PosixQueue> MqOpen(std::string name, int oflag, mode_t mode,
+                                struct mq_attr* attr) {
+  mqd_t fd = mq_open(name.c_str(), oflag, mode, attr);
+  if (fd == -1) {
+    return PosixError(errno, absl::StrFormat("mq_open(%d)", oflag));
+  }
+  return PosixQueue(fd, name);
+}
+
+// MqUnlink wraps mq_unlink(2).
+PosixError MqUnlink(std::string name) {
+  int err = mq_unlink(name.c_str());
+  if (err == -1) {
+    return PosixError(errno, absl::StrFormat("mq_unlink(%s)", name.c_str()));
+  }
+  return NoError();
+}
+
+// MqClose wraps mq_close(2).
+PosixError MqClose(mqd_t fd) {
+  int err = mq_close(fd);
+  if (err == -1) {
+    return PosixError(errno, absl::StrFormat("mq_close(%d)", fd));
+  }
+  return NoError();
+}
+
+// Test simple opening and closing of a message queue.
+TEST(MqTest, Open) {
+  SKIP_IF(IsRunningWithVFS1());
+
+  PosixQueue queue = ASSERT_NO_ERRNO_AND_VALUE(
+      MqOpen(O_RDWR | O_CREAT | O_EXCL, 0777, nullptr));
+}
+
+// Test mq_open(2) after mq_unlink(2).
+TEST(MqTest, OpenAfterUnlink) {
+  GTEST_SKIP();
```

Comment:
Why is this skipped unconditionally?

Reply:
Done. I generally add `GTEST_SKIP` at the start of each test when tests are first created. This is done mainly so the tests won't fail if somebody decides to checkout a random commit from the PR. I most likely forgot to remove it after implementing the stubs.